component/admin/models/tax_rates.php

```diff
@@ -162,4 +162,36 @@ protected function getStoreId($id = '')
 
         return parent::getStoreId($id);
     }
+
+	/**
+	 * @param   string  $id
+	 *
+	 * @return  array
+	 *
+	 * @since   3.0.2
+	 */
+	public static function getShopperTax($id)
+	{
+		$result = [];
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true);
+
+		$query->select('shopper_group_id')
+			->from($db->qn('#__redshop_tax_shoppergroup_xref'))
+			->where($db->qn('tax_rate_id') . ' = ' . $db->q($id));
+
+		$shopperGroupIds = $db->setQuery($query)->loadColumn();
```

Comment:
@hoang-tran-redweb Please check the way using **\Redshop\DB\Tool::** instead of loadObject, loadObjectList ... because library has using transaction and more safety query